compiler/AST/symbol.cpp

```diff
@@ -1508,6 +1511,111 @@ void createInitStringLiterals() {
   initStringLiterals->retType = dtVoid;
   initStringLiterals->insertAtTail(new CallExpr(PRIM_RETURN, gVoid));
   stringLiteralModule->block->insertAtTail(new DefExpr(initStringLiterals));
+
+  initStringLiteralsEpilogue = initStringLiterals->getOrCreateEpilogueLabel();
+
+  if (fMinimalModules) {
+    return;
+  }
+
+  // accumulate the string/bytes and prepare to sort them
+  std::vector<std::pair<std::string, VarSymbol*>> literals;
+
+  for_alist(expr, stringLiteralModule->block->body) {
+    if (DefExpr* def = toDefExpr(expr)) {
+      if (VarSymbol* s = toVarSymbol(def->sym)) {
+        if (s->hasFlag(FLAG_CHAPEL_STRING_LITERAL) ||
+            s->hasFlag(FLAG_CHAPEL_BYTES_LITERAL)) {
+          Immediate* imm = s->immediate;
+          INT_ASSERT(imm);
+          literals.push_back({imm->to_string(), s});
+        }
+      }
+    }
+  }
+
+  // sort the strings/bytes so that they can be stored in a reliable order
+  std::sort(literals.begin(), literals.end());
```

Comment:
Is this order still reliable if you have a Chapel string and a bytes that are identical? I think they'll end up being represented by different immediates, but they'll compare equal. So the ordering will rely on the VarSymbol pointers, which is indeterministic.

Reply:
That's mostly true, but we have defined `std::less` for the VarSymbol pointers. See

https://github.com/chapel-lang/chapel/blob/c75946acea8a1f98acd6235cb09e72046baf8254/compiler/include/baseAST.h#L438-L448 (which is later invoked as `def_less_ast(VarSymbol)`

We have quite a few `std::map`s that rely on this to have reproducible ordering.

Reply:
Ah, thanks!